fetcher/tar.py

```diff
@@ -99,18 +101,26 @@ def unpack(self, downloadFirst=False):
         update = tarfile.open(name=self.workspacefile, mode='r:*')
```

Comment:
Instead of passing a file here you can replace with a file object. Here is how you would open the stream returned by dockerpy:

```python
import docker
import tarfile

cli = docker.Client()
ctnr = cli.create_container('busybox', 'true')
strm = cli.export(ctnr)
tar = tarfile.open(mode='r|*', fileobj=strm)
tar.list()
```

The important bit is the `r|*` (instead of `r:*`) mode which tells tarfile to not perform any `seek()` operations and therefore can operate on a stream